solcore/analytic_solar_cells/depletion_approximation.py

```diff
@@ -383,6 +395,52 @@ def bc(ya, yb):
     return out
 
 
+def get_J_sc_diffusion_green(xa, xb, g, D, L, y0, S, ph, side='top'):
+    xbL = (xb - xa) / L
+    crvel = S / D * L
+
+    # if L too low in comparison to junction width, avoid nan's
+    if xbL > 1.e2:
+        if side == 'top':
+            cadd = -2. * S / D * y0 / (1. + crvel) * np.exp(-xbL)
+        else:
+            cadd = -2. * S / D * y0 / (1. - crvel) * np.exp(-xbL)
+        cp = 1.
+
+        def fun(x):
```

Comment:
While I've used them, I am not a big fan of internal functions like this one (and the one below) because they cannot be tested. I would suggest you define them as top level functions, passing all the relevant arguments to them. 

Again, the point being improving the overall quality of solcore's code

Reply:
I agree. `Quad_vec` does not support the "arg" argument like the other functions of the scipy.integate package (although the feature has been requested and will be implemented in the future). My suggestion is using the `partial` method from the `functiontool` package.